### PR TITLE
#52: implement CSCH excel function...

### PR DESCRIPTION
**Pull request by jongleur1983**
#### What's this PR do?
This PR implements the CSCH function provided by excel that calculates the hyperbolic cosecant of an angle.
See https://support.office.com/en-us/article/CSCH-function-f58f2c22-eb75-4dd6-84f4-a503527f8eeb?ui=en-US&rs=en-US&ad=US for Microsofts documentation.

#### Where should the reviewer start?
It's a simple PR of the function and a few tests checking the correct calculation.

#### How should this be manually tested?
To be honest: no idea, I followed PR #74 to get how it works. I there's anything I could do better, please tell me.
#### Questions:
- Is there a blog post?
no.
- Does the knowledge base need an update?
don't think so: As far as I see there's no documentation about individual implemented functions in the wiki.
- Does this add new (C#) dependencies?
no.

- [ ] C# Code Review: @csreviewer
- [ ] Test Automation Review: @csreviewer

### CODE REVIEW

File: ClosedXML/Excel/CalcEngine/Functions/MathTrig.cs [hunk 1]
```diff
@@ -118,7 +118,7 @@ private static object Cosh(List<Expression> p)
 
         private static object Csch(List<Expression> p)
         {
-            if (p[0].Evaluate().Equals(0))
+            if (p[0].Evaluate().Equals(0.0))
```
Comment (igitur) [inline ClosedXML/Excel/CalcEngine/Functions/MathTrig.cs]: For floating point comparison, use `if (Math.Abs(p[0].Evaluate()) < Double.Epsilon)`

File: ClosedXML_Tests/Excel/CalcEngine/MathTrigTests.cs [hunk 2]
```diff
@@ -139,5 +140,37 @@ public void SumProduct()
                 Assert.Throws<NoValueAvailableException>(() => ws.Evaluate("SUMPRODUCT(A1:A10, B1:B5)"));
             }
         }
+
+        [TestCase(1, 0.850918128)]
```
Comment (igitur) [inline ClosedXML_Tests/Excel/CalcEngine/MathTrigTests.cs]: For the tests, please follow the same pattern as the testing of other functions, where possible, i.e. try to use ` XLWorkbook.EvaluateExpr`

File: ClosedXML/Excel/CalcEngine/Functions/MathTrig.cs [hunk 3]
```diff
@@ -115,6 +116,15 @@ private static object Cosh(List<Expression> p)
             return Math.Cosh(p[0]);
         }
 
+        private static object Csch(List<Expression> p)
+        {
+            if (p[0].Evaluate().Equals(0.0))
+            {
+                return ErrorExpression.ExpressionErrorType.DivisionByZero;
```
Comment (igitur) [inline ClosedXML/Excel/CalcEngine/Functions/MathTrig.cs]: Should be `throw new DivisionByZeroException();`

**Comment by igitur** [comment]
Thanks for the contribution. Please have a look at the requested changes. They're minor. 

Also, if you want to have a stab at the other trigonometry functions, that would be great.

File: ClosedXML/Excel/CalcEngine/Functions/MathTrig.cs (hunk 1 above)
Comment (jongleur1983) [reply ClosedXML/Excel/CalcEngine/Functions/MathTrig.cs]: I agree in principle, but for 0 this should be save. If you want I'll change it of course.

File: ClosedXML_Tests/Excel/CalcEngine/MathTrigTests.cs (hunk 2 above)
Comment (jongleur1983) [reply ClosedXML_Tests/Excel/CalcEngine/MathTrigTests.cs]: yes. will do. Took the wrongest possible example for reference I guess, currently implementing DECIMAL with the other pattern already ;)

File: ClosedXML/Excel/CalcEngine/Functions/MathTrig.cs (hunk 1 above)
Comment (igitur) [reply ClosedXML/Excel/CalcEngine/Functions/MathTrig.cs]: You can't predict how the floating point rounding will work, e.g. `=CSCH(8-64/8)` . Please change it.

**Comment by igitur** [comment]
Please resolve the conflicts. Your other PR made changes to same file.

**Pull request merged**